gha: use sudo -E in some places to prevent dropping env-vars

thaJeztah opened the pull request:
Using `-E` preserves environment variables, except for PATH, so PATH has to be manually set to match the current environment.

I removed env-vars that were redundant (such as `GOPATH=$GOPATH`), which should be handled by `-E`.

thaJeztah commented:
Hmm.. interesting

```
CONTAINERD_RUNTIME=$TEST_RUNTIME make cri-integration
  shell: /bin/bash -e {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.15.5/x64
    GOPATH: /home/runner/work/containerd/containerd
    TEST_RUNTIME: io.containerd.runc.v2
+ bin/ctr
+ bin/containerd
+ bin/containerd-stress
+ bin/containerd-shim
+ bin/containerd-shim-runc-v1
+ bin/containerd-shim-runc-v2
+ binaries
+ bin/cri-integration.test
go test github.com/containerd/containerd/integration.test: copying /tmp/go-build334883615/b001/integration.test: open /home/runner/work/containerd/containerd/src/github.com/containerd/containerd/bin/cri-integration.test: permission denied
Error: Makefile:187: recipe for target 'bin/cri-integration.test' failed
make: *** [bin/cri-integration.test] Error 1
Error: Process completed with exit code 2.
```

estesp commented on .github/workflows/ci.yml:
```diff
@@ -278,8 +278,8 @@ jobs:
         run: |
-          make binaries
-          sudo make install
+          sudo -E PATH=$PATH make binaries
```
`make binaries` doesn't need to be run as root/via sudo

thaJeztah commented:
Updated; rebased, and removed the `sudo` for `make binaries`.


Looks like we may currently be running some tests on Go 1.15 currently; see https://github.com/containerd/containerd/pull/5362#issuecomment-822367638

Hopefully this fixes that

@estesp @mxpv @fuweid ptal

estesp reviewed (approved):
LGTM

dims reviewed (approved):
LGTM
